crates/ef-testing/src/sequencer/setup.rs

```diff
@@ -0,0 +1,263 @@
+    ) -> StateResult<()>;
+
+    fn fund(&mut self, evm_address: &Address, balance: U256) -> StateResult<()>;
+}
+
+impl InitializeEvmState for KakarotSequencer {
+    fn initialize_contract(
+        &mut self,
+        evm_address: &Address,
+        bytecode: &Bytes,
+        nonce: U256,
+        evm_storage: Vec<(U256, U256)>,
+    ) -> StateResult<()> {
+        let starknet_address = self.compute_starknet_address(evm_address);
+        let mut storage = vec![];
+
+        // Initialize the contract evm_address.
+        let evm_address: FeltSequencer = (*evm_address).into();
+        let evm_address_storage = (
+            get_storage_var_address("evm_address", &[]).unwrap(), // safe unwrap: var is ASCII
+            evm_address.into(),
+        );
+        storage.push(evm_address_storage);
+
+        // Initialize the is_initialized_ storage var.
+        let is_initialize_storage = (
+            get_storage_var_address("is_initialized_", &[]).unwrap(), // safe unwrap: var is ASCII
+            StarkFelt::from(1u8),
+        );
+        storage.push(is_initialize_storage);
+
+        // Initialize the contract owner
+        let owner_storage = (
+            get_storage_var_address("Ownable_owner", &[]).unwrap(), // safe unwrap: var is ASCII
+            contract_address_to_starkfelt(&KAKAROT_ADDRESS),
+        );
+        storage.push(owner_storage);
+
+        // Initialize the bytecode storage var.
+        let bytecode_storage = &mut split_bytecode_to_starkfelt(bytecode)
+            .into_iter()
+            .enumerate()
+            .map(|(i, bytes)| {
+                (
+                    get_storage_var_address("bytecode_", &[StarkFelt::from(i as u32)]).unwrap(), // safe unwrap: var is ASCII
+                    bytes,
+                )
+            })
+            .collect();
+        let bytecode_len_storage = (
+            get_storage_var_address("bytecode_len_", &[]).unwrap(), // safe unwrap: var is ASCII
+            StarkFelt::from(bytecode.len() as u32),
+        );
+        storage.append(bytecode_storage);
+        storage.push(bytecode_len_storage);
+
+        // Initialize the kakarot address.
+        let kakarot_address_storage = (
+            get_storage_var_address("kakarot_address", &[]).unwrap(), // safe unwrap: var is ASCII
+            contract_address_to_starkfelt(&KAKAROT_ADDRESS),
+        );
+        storage.push(kakarot_address_storage);
+
+        // Initialize the nonce storage var.
+        let nonce = StarkFelt::from(TryInto::<u128>::try_into(nonce).map_err(|err| {
+            StarknetApiError::OutOfRange {
+                string: err.to_string(),
+            }
+        })?);
+        let nonce_storage = (
+            get_storage_var_address("nonce", &[]).unwrap(), // safe unwrap: var is ASCII
+            nonce,
+        );
+        storage.push(nonce_storage);
+
+        // Initialize the storage vars.
+        let is_evm_storage_empty = evm_storage.is_empty();
```

Comment:
plz define var closer the usage

Reply:
is actually createing a variable necesary?

Reply:
I was using `into_iter` instead of `iter`, updated